### PR TITLE
Ensure polyfill works properly with Ember 3.27+ 

### PR DESCRIPTION
**Pull request by mydea**
Based on ef4's comment here: https://github.com/ember-polyfills/ember-cached-decorator-polyfill/issues/70#issuecomment-837176484

This PR changes how the polyfill works by simply rewriting `import { cached } from '@glimmer/tracking'` to `import { cached } from 'ember-cached-decorator-polyfill'`.

It uses a simple regex to do so - I added some tests covering all the different import scenarios I could think of (as `@glimmer/tracking` only has a single other relevant import `tracked`, it is relatively straightforward).

I hope I didn't miss anything critical :sweat_smile: but the tests are passing and it seems to be working just fine.

### CODE REVIEW

**Comment by mydea** [comment]
FYI, I have tried this in our medium-sized app with ember-source 3.27 (and TypeScript) and it worked fine!

**Comment by ef4** [comment]
Thanks for working on this.

My concern with this PR is that the regex-based rewriting looks pretty fragile. If `@glimmer/tracking` adds any new exported names, it's likely to break.

Instead of replacing the existing babel plugin with a regex, I think it would be better to modify the plugin so it does the rewriting of the imported path.

**Comment by mydea** [comment]
OK, I made a second try, now rewriting the import via the babel AST. I did not cover re-exporting it, but IMHO that seems OK to me. I did not 100% see through how I would make the re-export declaration work, I am not necessarily a babel-AST-pro :sweat_smile:

File: package.json [hunk 1]
```diff
@@ -39,13 +39,14 @@
     "test:ember-compatibility": "ember try:each"
   },
   "dependencies": {
+    "@glimmer/tracking": "^1.0.4",
+    "broccoli-replace": "^2.0.2",
```
Comment (rwjblue) [inline package.json:43]: I think this can be removed

**Review by rwjblue — CHANGES_REQUESTED** [review]
Thanks for picking this up! I left one minor comment (removing that new dependency that we don't need any more), but it should be good to go once that is fixed.

File: package.json (hunk 1 above)
Comment (mydea) [reply package.json]: Oops, totally forgot about that. Removed it!

**Comment by acorncom** [comment]
@mydea just pushed up a simple repro I'd built to test things as part of https://github.com/ember-polyfills/ember-cached-decorator-polyfill/issues/77 Ran it against your branch here and it seems to still be failing

**Comment by mydea** [comment]
True, I can see the same in our app. I will check it out and see if I can make it work..!

**Comment by mydea** [comment]
I can also reproduce it by updating the addon itself to ember-cli-babel@7.26.6 (which at least makes fixing it easier, I guess)

**Comment by mydea** [comment]
OK, seems from ember-cli-babel 7.23->7.24 the `_getEmberModulesAPIPolyfill` hook is dropped, leading to the whole babel plugin not running.

**Comment by mydea** [comment]
So, I have now refactored it _again_, it is much nicer now actually :D I am using ember-cli-babel-plugin-helpers to just add the plugin in a regular fashion, without patching anything. I have also updated ember-cli-babel in this addon itself to latest, and the tests are passing fine. Can @acorncom you try it again in the repro?

**Comment by acorncom** [comment]
Confirmed that it's no longer crashing in my repro and that the `@cache` decorator seems to be working smoothly

**Review by rwjblue — APPROVED** [review]
This is much better, thank you @mydea!

**Comment by mydea** [comment]
> Unfortunately, I am still seeing the error:
> 
> ```
> Uncaught TypeError: decorator is not a function
> ```
> 
> I'm pretty sure I had the latest ember-cached-decorator-polyfill installed, and worse is it failed even with ember-source@3.26.1
> 
> image

Hmm, I just tried it in our app:

ember-cached-decorator-polyfill@0.1.2
ember-source@3.26.1
ember-cli-babel@7.26.5

And I have tried it with:

ember-cached-decorator-polyfill@0.1.2
ember-source@3.27.1
ember-cli-babel@7.26.6

Both of these setups worked fine for me.

**Comment by nightire** [comment]
> Hmm, I just tried it in our app:
> 
> ember-cached-decorator-polyfill@0.1.2
> ember-source@3.26.1
> ember-cli-babel@7.26.5
> 
> And I have tried it with:
> 
> ember-cached-decorator-polyfill@0.1.2
> ember-source@3.27.1
> ember-cli-babel@7.26.6
> 
> Both of these setups worked fine for me.

I have tried these combinations as well but still no luck, I even use `resolutions` to constraint these versions, and the error still remains. I'm out of options now.😕

**Pull request merged**